### DISCUSSION TITLE
Custom widgets API

### DISCUSSION THREAD

**Discussion by Moosems** [tukaan/tukaan#98] [Ideas]
I noticed that you mentioned effbot.org in one of your repositories and when looking at it through the web archive I noticed they had created a Widget Creation Tool that they named WCK. I was wondering if we wanted to make widget creation easier in Tukaan and, if so, how can I help make this? I've always wanted something like this so seeing it happen in Tukaan would be awesome

**Comment by rdbende** [comment]
Cool idea! We definitely need some custom widget API for both megawidgets and plain widgets.

To be honest I've wondered several times what about incorporating some form of the WCK into Tukaan. For some reason I always dropped that idea. But now it occurs to me that we could make some kind of "wrapper" around the [Ttk C API](https://www.tcl-lang.org/man/tcl8.6/TkLib/contents.htm), so that not only "megawidgets" could be created, but full-fledged ttk widgets.

**Comment by Moosems** [comment]
How can I help with this?
